tests/integration/member_test.go

```diff
@@ -115,3 +119,52 @@ func TestSnapshotAndRestartMember(t *testing.T) {
 		}
 	}
 }
+
+func TestRemoveMember(t *testing.T) {
+	integration.BeforeTest(t)
+	c := integration.NewCluster(t, &integration.ClusterConfig{Size: 3, UseBridge: true, BackendBatchInterval: 1000 * time.Second})
+	defer c.Terminate(t)
+	time.Sleep(etcdserver.HealthInterval)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	_, err := c.Members[2].Client.MemberRemove(ctx, uint64(c.Members[0].Server.MemberId()))
+	cancel()
+
+	if err != nil {
+		t.Fatalf("should accept removing member: %s", err)
+	}
+	c.Members[0].Stop(t)
+	time.Sleep(etcdserver.HealthInterval)
+
+	checkMemberCount(t, c.Members[1], 2)
+	checkMemberCount(t, c.Members[2], 2)
+}
+
+func checkMemberCount(t *testing.T, m *integration.Member, expectedMemberCount int) {
+	memberCount := mustReadMemberCountFromBackend(t, m.Server.Backend())
+	if memberCount != expectedMemberCount {
+		t.Errorf("Expect MemberCountFromBackend=%d, got %d", expectedMemberCount, memberCount)
+	}
```

Comment:
This doesn't make much sense to me. bbolt not returning latest data doesn't mean etcd can't return the latest data. We are testing etcd instead of bbolt.

Reply:
This test is to test the assumption `MustReadMembersFromBackend` should return the correct members, like in https://github.com/etcd-io/etcd/issues/17052. 
Refactored to make it more clear.